### PR TITLE
SwitchStatus method and tests

### PR DESCRIPTION
**Pull request by chrisweber1408**
checked in my switchStatus method in the backend if admin is logged in

### CODE REVIEW

File: backend/src/main/java/com/example/dontwastetomuch/game/GameController.java [hunk 1]
```diff
@@ -31,28 +32,34 @@ public List<Game> getAllGames(){
     }
 
     @GetMapping("/{gameId}")
-    public UserGameDTO getOneGame(@PathVariable String gameId, Principal principal){
+    public UserGameDTO getOneOfMyGames(@PathVariable String gameId, Principal principal){
         MyUser myUser = myUserRepo.findById(principal.getName()).orElseThrow();
         UserGameDTO userGameDTO = new UserGameDTO();
```
Comment (jekattack) [inline backend/src/main/java/com/example/dontwastetomuch/game/GameController.java:37]: Using the builder-Pattern could save you some lines

File: backend/src/main/java/com/example/dontwastetomuch/game/GameService.java [hunk 2]
```diff
@@ -24,19 +24,25 @@ public List<Game> getAllGames() {
...
-    public void switchStatus(Game game) {
-        if (game.isApproved()) {
-            game.setApproved(false);
-            gameRepo.save(game);
-        }else {
-            game.setApproved(true);
-            gameRepo.save(game);
+    public void switchStatus(Game game, MyUser user) {
+        if (!user.getRoles().stream().anyMatch(roles -> roles.contains("admin"))){
+            throw new IllegalArgumentException("No admin logged in");
```
Comment (jekattack) [inline backend/src/main/java/com/example/dontwastetomuch/game/GameService.java:34]: You can handle restrictions for diffrent roles in the SecurityConfig. As long as you wont call this method in any unrestricted context, you can save this if-statement.

**Review by jekattack — APPROVED** [review]
lgtm
tl;dr:
- I suggest the builderpattern
- Maybe move code to service
- Maybe leave the role-handling to SecurityConfig
